include/marl/pool.h

```diff
@@ -275,51 +305,9 @@ BoundedPool<T, N, POLICY>::BoundedPool(
     Allocator* allocator /* = Allocator::Default */)
     : storage(allocator->make_shared<Storage>(allocator)) {}
 
-template <typename T, int N, PoolPolicy POLICY>
-typename BoundedPool<T, N, POLICY>::Loan BoundedPool<T, N, POLICY>::borrow()
-    const {
-  Loan out;
-  borrow(1, [&](Loan&& loan) { out = std::move(loan); });
-  return out;
-}
-
-template <typename T, int N, PoolPolicy POLICY>
-template <typename F>
-void BoundedPool<T, N, POLICY>::borrow(size_t n, const F& f) const {
-  marl::lock lock(storage->mutex);
-  for (size_t i = 0; i < n; i++) {
-    storage->returned.wait(lock, [&] { return storage->free != nullptr; });
-    auto item = storage->free;
-    storage->free = storage->free->next;
-    if (POLICY == PoolPolicy::Reconstruct) {
-      item->construct();
-    }
-    f(std::move(Loan(item, storage)));
-  }
-}
-
-template <typename T, int N, PoolPolicy POLICY>
-std::pair<typename BoundedPool<T, N, POLICY>::Loan, bool>
-BoundedPool<T, N, POLICY>::tryBorrow() const {
-  Item* item = nullptr;
-  {
-    marl::lock lock(storage->mutex);
-    if (storage->free == nullptr) {
-      return std::make_pair(Loan(), false);
-    }
-    item = storage->free;
-    storage->free = storage->free->next;
-    item->pool = this;
-  }
-  if (POLICY == PoolPolicy::Reconstruct) {
-    item->construct();
-  }
-  return std::make_pair(Loan(item, storage), true);
-}
-
 template <typename T, int N, PoolPolicy POLICY>
 void BoundedPool<T, N, POLICY>::Storage::return_(Item* item) {
-  if (POLICY == PoolPolicy::Reconstruct) {
+  if constexpr (POLICY == PoolPolicy::Reconstruct) {
```

Comment:
Marl is currently C++ 11, `if constexpr` is C++ 17.

Reply:
Unresolving, as the `constexpr` are still here.

Reply:
fixed